lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,421 @@
+            lambda tb: tickmock("tarball_md5", "my_md5_string"),
+        )
+        obj = UnpackTarballs(server_config, make_logger)
+        obj.unpack(self.tar, MockFileTree())
+        assert mocks_called == ["tarball_md5", "unpack"]
+
+    def test_unpack_symlinks_creation(
+        self, monkeypatch, server_config, make_logger, caplog
+    ):
+        """Show that when symlink creation fails and raises
+        FileNotFoundError Exception it is handled successfully."""
+        mocks_called = []
+
+        def mock_symlink(path: Path, dest: Path):
+            """
+            Mock of os.symlink() function
+
+            Args:
+                path: tarball file path
+                dest: destination path to create symlink
+
+            Returns:
+                Raises FileNotFoundError Exception
+            """
+            mocks_called.append("mock_symlink")
+            raise FileNotFoundError(f"'{path}' -> '{dest}'")
+
+        linkedlistdest = "TO-INDEX"
+        monkeypatch.setattr(os, "symlink", mock_symlink)
+        obj = UnpackTarballs(server_config, make_logger)
+        with pytest.raises(FileNotFoundError):
+            obj.update_symlink(self.tar)
+            assert (
+                f"Error in creation of symlink. '{server_config.ARCHIVE / 'ABC' / self.tar.name}' -> '{server_config.ARCHIVE / 'ABC' / linkedlistdest / self.tar.name}'"
+                in caplog.text
+            )
+        assert mocks_called == ["mock_symlink"]
+
+    def test_unpack_symlinks_removal(
+        self, monkeypatch, server_config, make_logger, caplog
+    ):
+        """Show that, when unlinking the tarball results in
+        a FileNotFoundError, it is handled successfully."""
+        mocks_called = []
+
+        def tickmock(id: str, ret_val: Any) -> Any:
+            mocks_called.append(id)
+            return ret_val
+
+        def mock_unlink(path):
+            """
+            Mock of os.unlink() function
+
+            Args:
+                path: tarball path in 'TO-INDEX' state directory
+
+            Returns:
+                Raises FileNotFoundError Exception
+            """
+            mocks_called.append("mock_unlink")
+            raise FileNotFoundError(f"No such file or directory: '{path}'")
+
+        monkeypatch.setattr(os, "symlink", lambda path, dest: tickmock("symlink", None))
+        monkeypatch.setattr(os, "unlink", mock_unlink)
+        obj = UnpackTarballs(server_config, make_logger)
+        with pytest.raises(FileNotFoundError):
+            obj.update_symlink(self.tar)
+            assert (
+                f"Error in removing symlink from TO-UNPACK state. No such file or directory: '{server_config.ARCHIVE / 'ABC' / obj.LINKSRC / self.tar.name}'"
+                in caplog.text
+            )
+        assert mocks_called == ["symlink", "symlink", "mock_unlink"]
+
+    def test_unpack_symlinks_success(
+        self, monkeypatch, server_config, make_logger, caplog
+    ):
```

Comment:
I don't think we're using caplog in this test.

Ditto `test_unpack_tarballs_success()`.